packages/core/src/indexing-store/utils/sort.ts

```diff
@@ -0,0 +1,59 @@
+import type { Schema } from "@/schema/types.js";
+import { isBaseColumn, isEnumColumn } from "@/schema/utils.js";
+import type { OrderByInput } from "../store.js";
+
+export type OrderByConditions = [
+  columnName: string,
+  direction: "asc" | "desc",
+][];
+
+export function buildOrderByConditions({
+  orderBy,
+  table,
+}: {
+  orderBy: OrderByInput<any> | undefined;
+  table: Schema["tables"][keyof Schema["tables"]];
+}): OrderByConditions {
+  if (!orderBy) {
+    return [["id", "asc"]];
+  }
+
+  const conditions = Object.entries(orderBy);
+  if (conditions.length > 1)
+    throw new Error("Invalid sort. Cannot sort by multiple columns.");
+
+  const [columnName, orderDirection] = conditions[0];
+
+  // Validate column name.
+  const column = table[columnName];
+  if (!column) {
+    throw Error(
+      `Invalid sort. Column does not exist. Got '${columnName}', expected one of [${Object.keys(
+        table,
+      )
+        .filter((key) => isBaseColumn(table[key]) || isEnumColumn(table[key]))
+        .map((c) => `'${c}'`)
+        .join(", ")}]`,
+    );
+  }
+  if (column._type === "m" || column._type === "o") {
+    throw Error(
+      `Invalid sort. Cannot filter on virtual column '${columnName}'.`,
+    );
+  }
+
+  if (orderDirection === undefined || !["asc", "desc"].includes(orderDirection))
```

Comment:
First expression isn't needed

Reply:
Is convenient for type checking on line 50